### PR TITLE
😵 Limit faucet harassing 

### LINKED ISSUE #112: 😵 Discord bot faucet harassing
## Issue

We're seeing abuse of the discord bot to obtain testnet (nemeton) tokens, which is leading to high pressure on the service, resulting in outages and starvation.

As a preventive measure, the faucet was stopped.

## Idea

Implement a simple and effective protection mechanism that limits the number of requests submitted per community member per unit of time.

### PR DESCRIPTION
**Pull request by bdeneux**
#### 📝 Description 

Due to the faucet message harassing, it was necessary to improve message handling to avoid spamming. The solution used in this PR bring a new messages queue linked to the sender, on each transaction batch, the first N transaction message (config set in config file) is sent with the discord message. The queue contains the list of all transaction pending, if user try to request funds and already waiting in the queue, the transaction is skipped. The queue is also limited in size, if the size limit is reached, transaction is skipped also. 

New error messages has been configured to be sent to the discord user if the queue is full or if the user already request funds. 

#### ⚙️ Configuration

New configuration keys has been added to configure the maximum number of message that can be sent by transaction batch, by default I set it to 7 transactions and the maximum number of transaction that can be register in the waiting queue, by default 1000 transactions. 

```toml
[chain]
max_msg = 7
queue_limit = 1000
``` 

#### 🦀 Rust update

I've also update the rust toolchain to the latest version 1.65. It's allow me to use the new [if-else statement](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html#let-else-statements) that is wonderful 😇.  https://github.com/okp4/discord-bot/pull/113/commits/297fc6e8b48f39e5d0a6060056482e7db77380a8

Closes #112

### CODE REVIEW

**Review by ingvaar — APPROVED** [review]
LGTM

File: src/discord/discord_server/cmd/request.rs [hunk 1]
```diff
@@ -29,11 +32,21 @@ impl CommandExecutable for RequestCmd {
...
+                .send(RequestFunds {
+                    address,
+                    requester: command.user.clone(),
+                })
+                .await;
+            match request.map_err(|e| Mailbox(e.to_string())).and_then(|r| r) {
+                Ok(_) => "📥 Funds has been successfully requested.".to_string(),
+                Err(err) => match err {
+                    RequestError::QueueFull => "❌ Queue is full, please wait before re-submit funds request.".to_string(),
+                    RequestError::DuplicateUser => "❌ You have already request funds, please wait before re-submit your request.".to_string(),
```
Comment (ad2ien) [inline src/discord/discord_server/cmd/request.rs:46]:
```suggestion
                    RequestError::DuplicateUser => "❌ You have already requested funds, please wait before re-submiting your request.".to_string(),
```

**Review by ccamel — APPROVED** [review]
LGTM!

**Review by ad2ien — APPROVED** [review]
LGTM 👏👏

**Review by amimart — APPROVED** [review]
Ok let's try this strategy 😉

**Pull request merged**